plugins/woocommerce/includes/tracks/class-wc-site-tracking.php

```diff
@@ -93,6 +93,16 @@ public static function add_tracking_function() {
 					delete( eventProperties._ui );
 					delete( eventProperties._ut );
 				}
+				// Verify the event name is correct
+				if ( ! /^(([a-z0-9]+)_){2}([a-z0-9_]+)$/.test( eventName ) ) {
```

Comment:
Could we add some tests around this?

Reply:
This also does not allow single property names.  Not sure if that's intentional or not, but I think they're allowed, right?

```js
window.wcTracks.recordEvent( 'test', { 'bad-property': 'mooo' } )
```

Reply:
> This also does not allow single property names. Not sure if that's intentional or not, but I think they're allowed, right?

I thought they weren't. I used the same regex declared [here](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/tracks/class-wc-tracks-event.php#L17-L25). After a quick search throughout the code, I couldn't find any single property name for the event name, I also tried to send to Tracks a single property name as the event name, but the event hasn't been recorded.

In order to have a single source of truth, I added the commit 8ca585e to get the regex from the class `WC_Tracks_Event`.

Reply:
> I also tried to send to Tracks a single property name as the event name, but the event hasn't been recorded.

Did it show up under rejected or not at all?  What was the event name you used.

>I thought they weren't. I used the same regex declared [here](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/includes/tracks/class-wc-tracks-event.php#L17-L25).

Okay, we won't block this since it matches our existing logic, but looks like @psealock was the last to work on that so pinging him in case he has any further context.

Reply:
> Did it show up under rejected or not at all? What was the event name you used.

@joshuatf Yes, it does. The event name I used was `singleeventname`. I just tried again so it should show up (as rejected) if you look for it in the Live Tracks tool.